### DISCUSSION TITLE
Express Route peering subnets are missing

### DISCUSSION THREAD

**Discussion by cata008** [Azure/ipam#175] [Ideas]
Hi @DCMattyG,

Anothering thing that I noticed during my tests is that Express Route peering subnets are not being scanned by IPAM.

It basically has to scan the existing ER circuit peerings and retrieve the primary and secondary subnets that were used to establish the connection.

Cheers,
Cata

**Comment by DCMattyG** [comment]
Hey @cata008, can you help me understand this a bit better? Are we talking about vNETs/Subnets on two sides on an ExpressRoute?

I'm just not sure what's missing here as I don't have an ExpressRoute in my test environment. Any additional details and/or screenshots will help immensely. Thanks so much!

**Reply by cata008** [reply]
Hi @DCMattyG,

You can find more info in the article below:
https://learn.microsoft.com/en-us/azure/expressroute/expressroute-routing#ip-addresses-used-for-peering

In our case, we used two /30 private address spaces. They can be found in the configuration of the express route circuit peering.

$er = Get-AzExpressRouteCircuit -Name erCircuitName
$erCircuitPeeringConfig = Get-AzExpressRouteCircuitPeeringConfig -ExpressRouteCircuit $er
$erCircuitPeeringConfig.PrimaryPeerAddressPrefix
$erCircuitPeeringConfig.SecondaryPeerAddressPrefix

We only have an AzurePrivatePeering configured.
![image](https://github.com/Azure/ipam/assets/37331541/463d18be-64e9-4056-a725-ff07a806ec95)

Hope it helps,
Cata

**Reply by StefanIvemo** [reply]
Just out of curiosity. What is the reason for having these subnets discovered by IPAM? They are public IPs used for the BGP session.

**Reply by cata008** [reply]
@StefanIvemo, if you'd checked the article that I pasted, you'd have seen that: "_You can use either private IP addresses or public IP addresses to configure the peering. The address range used for configuring routes can't overlap with address ranges used for virtual networks in Azure._"

**Reply by cata008** [reply]
@DCMattyG, do you need additional details or things are clear now? Cheers!

**Reply by DCMattyG** [reply]
Hey @cata008, I have what I need for now. Thanks so much for the level of detail. I'll get working on this right away 👍
